crypto/evp/evp_locl.h

```diff
@@ -95,3 +112,23 @@ void *evp_generic_fetch(OPENSSL_CTX *ctx, int operation_id,
                                             OSSL_PROVIDER *prov),
                         int (*upref_method)(void *),
                         void (*free_method)(void *));
+
+OSSL_PARAM *evp_pkey_to_param(EVP_PKEY *pkey, size_t *sz);
+
+#define M_check_autoarg(ctx, arg, arglen, err) \
+    if (ctx->pmeth->flags & EVP_PKEY_FLAG_AUTOARGLEN) {           \
+        size_t pksize = (size_t)EVP_PKEY_size(ctx->pkey);         \
+                                                                  \
+        if (pksize == 0) {                                        \
+            EVPerr(err, EVP_R_INVALID_KEY); /*ckerr_ignore*/      \
+            return 0;                                             \
```

Comment:
Flow of control in a macro :(
Might this be better as a function?

I know it's old code being moved...

Reply:
I'd rather not change that within this PR. The "return" from within the macro means that a simple replace with a function isn't possible - all call sites would need some tweaking. It starts to stray a little far from the scope of this PR.